File: aoj/volume100/python3/A10013.py
import sys

for line in sys.stdin:
    h, w = map(int, line.rstrip().split())
    if h == 0 and w == 0:
        break

    edge = '#'*w + '\n'
    rect = ''.join([edge, ('#' + '.'*(w-2) + '#\n') * (h-2), edge])

    print(rect)

File: aoj/volume100/nodejs/A10013.js
function main(lines) {
  for (var i = 0;; i++) {
    if (lines[i] == '0 0')
      break;
    
    var nums = lines[i].split(/\s+/);
    var h = parseInt(nums[0], 10);
    var w = parseInt(nums[1], 10);

    var edge   = repeat('#', w) + '\n';
    var inside = '#' + repeat('.', w-2) + '#\n'
    var rect   = [edge, repeat(inside, h-2), edge].join('');
    console.log(rect);
  }
}

function repeat(s, count) {
  return new Array(count+1).join(s);
}

var chunks = [];
process.stdin.on('data', function(chunk) {
  chunks.push(chunk)
});
process.stdin.on('end', function() {
  var lines = chunks.join('').split('\n');
  main(lines);
});
